src/meta/src/manager/catalog/mod.rs

```diff
@@ -836,66 +838,296 @@ where
+
+        // 3. update all relations that depend on this table, note that indexes are not included.
+        self.alter_relation_name_refs_inner(
+            database_core,
+            table_id,
+            &old_name,
+            table_name,
+            vec![table],
+            vec![],
+            vec![],
+            source,
+        )
+        .await
+    }
+
+    // TODO: refactor dependency cache in catalog manager for better performance.
+    #[allow(clippy::too_many_arguments)]
+    async fn alter_relation_name_refs_inner(
+        &self,
+        database_mgr: &mut DatabaseManager,
+        relation_id: RelationId,
+        from: &str,
+        to: &str,
+        mut to_update_tables: Vec<Table>,
+        mut to_update_views: Vec<View>,
+        mut to_update_sinks: Vec<Sink>,
+        to_update_source: Option<Source>,
+    ) -> MetaResult<NotificationVersion> {
+        for table in database_mgr.tables.values() {
+            if table.dependent_relations.contains(&relation_id) {
+                let mut table = table.clone();
+                table.definition = alter_relation_rename_refs(&table.definition, from, to);
+                to_update_tables.push(table);
+            }
+        }
+
+        for view in database_mgr.views.values() {
+            if view.dependent_relations.contains(&relation_id) {
+                let mut view = view.clone();
+                view.sql = alter_relation_rename_refs(&view.sql, from, to);
+                to_update_views.push(view);
+            }
+        }
+
+        for sink in database_mgr.sinks.values() {
+            if sink.dependent_relations.contains(&relation_id) {
+                let mut sink = sink.clone();
+                sink.definition = alter_relation_rename_refs(&sink.definition, from, to);
+                to_update_sinks.push(sink);
+            }
+        }
+
+        // commit meta.
+        let mut tables = BTreeMapTransaction::new(&mut database_mgr.tables);
+        let mut views = BTreeMapTransaction::new(&mut database_mgr.views);
+        let mut sinks = BTreeMapTransaction::new(&mut database_mgr.sinks);
+        let mut sources = BTreeMapTransaction::new(&mut database_mgr.sources);
+        to_update_tables.iter().for_each(|table| {
+            tables.insert(table.id, table.clone());
+        });
+        // TODO: there are some inconsistencies in the process of notifying the frontend, we need to
+        // support batch notification.
+        to_update_views.iter().for_each(|view| {
+            views.insert(view.id, view.clone());
+        });
+        to_update_sinks.iter().for_each(|sink| {
+            sinks.insert(sink.id, sink.clone());
+        });
+        if let Some(source) = to_update_source {
+            sources.insert(source.id, source);
+        }
+        commit_meta!(self, tables, views, sinks, sources)?;
+
+        // 5. notify frontend.
+        let mut version = 0;
```

Comment:
We may also check not all `to_update` are empty so the final version will not be accidentally `0`.